### PR TITLE
o/devicestate: only add track validation-sets task in run-mode.

### PR DESCRIPTION
**Pull request by Meulengracht**
Before it would result it being added to install mode as well, when we only set up the minimal set of snaps (gadget, kernel, base, snapd). If the validation-set contained any snaps besides these, it would fail. Wait until run mode, where we set up any non-essential snaps, and after this we expect all snaps to be present in system.

### CODE REVIEW

File: tests/nested/manual/core20-validation-sets/asserts/core-20.json
Comment (pedronis) [inline tests/nested/manual/core20-validation-sets/asserts/core-20.json:1]: nitpick: this should be called core-20-model.json maybe

File: tests/nested/manual/core20-validation-sets/asserts/vs.json
Comment (pedronis) [inline tests/nested/manual/core20-validation-sets/asserts/vs.json:1]: nitpick: bar-vs.json

File: overlord/devicestate/firstboot20_test.go [hunk 1]
```diff
@@ -1279,6 +1292,46 @@ func (s *firstBoot20Suite) TestPopulateFromSeedCore20ValidationSetTrackingHappy(
...
+				"presence": "required",
+				"revision": "1",
+			},
+			map[string]interface{}{
+				"name":     "pc",
+				"id":       s.AssertedSnapID("pc"),
+				"presence": "required",
+				"revision": "1",
+			},
+		},
```
Comment (pedronis) [inline overlord/devicestate/firstboot20_test.go:1316]: for increased realism shouldn't the validation set include a snap that is not installed in install mode? same for the run mode test

**Review by pedronis — COMMENTED** [review]
thank you, some small comments. I haven't looked very closely at the nested.sh changes

File: tests/nested/manual/core20-validation-sets/asserts/core-20.json
Comment (Meulengracht) [reply tests/nested/manual/core20-validation-sets/asserts/core-20.json:1]: thanks for suggestions here, fixed

File: tests/nested/manual/core20-validation-sets/asserts/vs.json
Comment (Meulengracht) [reply tests/nested/manual/core20-validation-sets/asserts/vs.json:1]: fixed

File: overlord/devicestate/firstboot20_test.go (hunk 1 above)
Comment (Meulengracht) [reply overlord/devicestate/firstboot20_test.go]: Super good point, they definitely missed that - it's added now

**Review by pedronis — APPROVED** [review]
thank you

File: tests/lib/nested.sh [hunk 2]
```diff
@@ -703,6 +703,36 @@ nested_prepare_base() {
...
+    nested_prepare_gadget
+    nested_prepare_base
+}
+
+nested_configure_default_user() {
+    local IMAGE_NAME
+    IMAGE_NAME="$(nested_get_image_name core)"
+    # Configure the user for the vm
+    if [ "$NESTED_USE_CLOUD_INIT" = "true" ]; then
+        if nested_is_core_20_system || nested_is_core_22_system; then
```
Comment (ernestl) [inline tests/lib/nested.sh:723]: consider using nested_get_version >= 20

**Review by ernestl — APPROVED** [review]
Looks good

**Pull request merged**